### PR TITLE
Add unicode_pair parameter.

### PR DESCRIPTION
**Pull request by eminbugrasaral**
## Example

     >>> slugify.sluggify(u'This is \xe9 test', unicode_pairs={u'\xe9', 'a'})
    u'this-is-a-test'
    instead of u'this-is-\xe9-test'

### CODE REVIEW

File: slugify/__init__.py [hunk 1]
```diff
@@ -39,4 +39,11 @@ def slugify(s, ok=SLUG_OK, lower=True, spaces=False):
     new = ''.join(rv).strip()
     if not spaces:
         new = re.sub('[-\s]+', '-', new)
-    return new.lower() if lower else new
+
+    new = new.lower() if lower else new
+
+    if unicode_pairs and isinstance(unicode_pairs, dict):
```
Comment (davedash) [inline slugify/__init__.py]: Can be rewritten as:

`if isinstance(unicode_pairs, dict)`

**Comment by davedash** [comment]
So I'm not 100% sure I understand the reasoning for this.

It almost seems like you want to do some overrides, but I'm not sure if this belongs in the slugify tool or if it should exist outside this method.

e.g. (not very efficient, but drives the point)

```
str = "my unicode string"
for k,v in replacement_dict.iteritems():
   str = str.replace(k, v)
slugify(str)
```

**Comment by eminbugrasaral** [comment]
Sometimes you want the output to neglect some unicode chars but you actually wanna hold the letter with a representation because of searching comp abilities with different languages. You can see my fork as an example: https://github.com/eminbugrasaral/unicode-slugify-turkish

**Comment by davedash** [comment]
What happens now?  They get turned to ‘-‘?  It seems like your turkish replacement should be the default (in fact, I have a hard time seeing where I wouldn’t want turkish letters replaced).




What are your thoughts?  I realize that changes the direction of this pull request a bit, but hopefully in a direction that benefits you.

**Comment by eminbugrasaral** [comment]
There are more letters than the common ones so I thought if we enable an extra parameter as a pair of unicode char and its representation dictionary, people can replace any char if needed. Turkish alphabet does not have 'â', but people who typed 'a' should be able to match something like: 'âpple'

**Comment by eminbugrasaral** [comment]
By the way, it's like this:

    str = "my unicode string"
    slug = slugify(str)
    for k,v in replacement_dict.iteritems():
        slug = slug.replace(k, v)

**Comment by eminbugrasaral** [comment]
We can also switch this to a boolean like 'smart_replace' to replace all common letters in the alphabet. It's up to you.

**Pull request closed without merging**